contracts/merger/TribeRagequit.sol

```diff
@@ -0,0 +1,106 @@
+    address public constant guardian = 0xB8f482539F2d3Ae2C9ea6076894df36D1f632775;
+
+    uint256 public intrinsicValueExchangeRateBase;
+
+    address public constant rewardsDripper = 0x3Fe0EAD3500e767F0F8bC2d3B5AF7755B1b21A6a;
+    IFei public constant fei = IFei(0x956F47F50A910163D8BF957Cf5846D573E7f87CA);
+
+    uint256 public immutable rageQuitEnd;
+
+    mapping(address => uint256) public claimed;
+
+    event Exchange(address indexed from, uint256 amountIn, uint256 amountOut);
+
+    bytes32 public merkleRoot;
+
+    constructor(
+        bytes32 root, 
+        uint256 _rageQuitEnd,
+        address tribeRariDAO
+    ) MergerBase(tribeRariDAO) {
+        merkleRoot = root;
+
+        rageQuitEnd = _rageQuitEnd;
+    }
+
+    /// @notice ragequit held TRIBE with FEI
+    /// @dev not gonna make it
+    /// @param amount the amount to redeem in TRIBE
+    /// @param key the amount of TRIBE allocated to the caller in the merkle drop
+    /// @param merkleProof a proof proving that the caller may redeem up to `key` amount of tribe
+    function ngmi(
+        uint256 amount,
+        uint256 key,
+        bytes32[] memory merkleProof
+    ) public {
+        require(isEnabled(), "Proposals are not both passed");
+        require(block.timestamp < rageQuitEnd, "outside ragequit window");
+        address thisSender = msg.sender;
+        require(
+            verifyClaim(thisSender, key, merkleProof),
+            "invalid proof"
+        );
+        require(
+            (claimed[thisSender] + amount) <= key,
+            "already ragequit all you tokens"
+        );
+        claimed[thisSender] = claimed[thisSender] + amount;
+        uint256 tribeTokenTakenTotal = amount;
+        uint256 token1GivenTotal = amount * intrinsicValueExchangeRateBase / scalar;
+        tribe.safeTransferFrom(thisSender, coreAddress, tribeTokenTakenTotal);
+        fei.mint(thisSender, token1GivenTotal);
+        emit Exchange(thisSender, tribeTokenTakenTotal, token1GivenTotal);
+    }
+
+    function getCirculatingTribe() public view returns (uint256) {
+        return tribe.totalSupply() - tribe.balanceOf(coreAddress) - tribe.balanceOf(rewardsDripper);
+    }
+    
+    /// @notice recalculate the exchange amount using the protocolEquity
+    /// @param protocolEquity the protocol equity
+    /// @return the new intrinsicValueExchangeRateBase
+    function exchangeRate(uint256 protocolEquity) public view returns (uint256) {
+        return (scalar * uint256(protocolEquity)) / getCirculatingTribe();
+    }
+
+    /// @notice Update the exchange rate based on protocol equity
+    /// @param protocolEquity the protocol equity
+    /// @return the new exchange rate
+    /// @dev only callable once by guardian
+    function setExchangeRate(uint256 protocolEquity) public returns (uint256) {
+        require(intrinsicValueExchangeRateBase == 0, "already set");
+        require(msg.sender == guardian, "guardian");
```

Comment:
remove and change to modifier `onlyGuardian`

Reply:
This isn't CoreRef. I think its fine as-is